src/common/connection.ts

```diff
@@ -543,7 +573,7 @@ class Connection extends EventEmitter {
       // JSON.stringify automatically removes keys with value of 'undefined'
       const message = JSON.stringify(Object.assign({}, request, {id}))
 
-      this._whenReady(this._send(message))
+      this._send(message)
```

Comment:
There's no guarantee here that this class is ready. Shouldn't you `await _whenReady()` above here?

Reply:
+1, good catch!

Reply:
Ah wait no, this is right. (it's been so long since I originally wrote this, I had to go back and investigate :) 

whenReady wasn't working as expected, so before this PR this was essentially a no-op (not waiting for anything). It's a bit weird, but what was basically happening was:

1.  `this._whenReady` won't resolve until the first "subscribe" request completes. So if you await it inside the request function itself, you've essentially bricked the request method.
2. Luckily (unluckily?) the way we were calling `this._whenReady` before this PR wasn't actually doing any waiting. Since `this._send(message)` was passed as an argument, JS would have to run the method first (making the actual request) and then pass the result as the `_whenReady()` argument. So whenReady was always being called after the request.
3. This PR fixes that bug in other places, but just removes the no-op call here since it bricks the current connection flow. If this is new functionality that we actually do want, I'd like to add it in a separate PR, given how much is already in flight at the moment.

```
// Called second, with result from this._send() as argument.
this._whenReady(
  // Called first.
  this._send(message)
);
```

Reply:
Oh interesting. Thank you for the explanation.